pkg/api/v20220904/clustermanager.go

```diff
@@ -0,0 +1,198 @@
+	// This is a flag used during the swagger generation typewalker to
+	// signal that it should be marked as a proxy resource and
+	// not a tracked ARM resource.
+	proxyResource bool
+
+	// The Resource ID.
+	ID string `json:"id,omitempty"`
+
+	// The resource name.
+	Name string `json:"name,omitempty"`
+
+	// The resource type.
+	Type string `json:"type,omitempty" mutable:"case"`
+
+	// SystemData metadata relating to this resource.
+	SystemData *SystemData `json:"systemData,omitempty"`
+
+	// The MachinePool Properties
+	Properties MachinePoolProperties `json:"properties,omitempty"`
+}
+
+// MachinePoolProperties represents the properties of a MachinePool
+type MachinePoolProperties struct {
+	Resources string `json:"resources,omitempty"`
+}
+
+// SyncSetList represents a list of SyncSets
+type SyncIdentityProviderList struct {
+	// proxyResource bool
+
+	// The list of syncsets.
+	SyncIdentityProviders []*SyncIdentityProvider `json:"value"`
+
+	// The link used to get the next page of operations.
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+// SyncIdentityProvider represents a SyncIdentityProvider
+type SyncIdentityProvider struct {
+	// This is a flag used during the swagger generation typewalker to
+	// signal that it should be marked as a proxy resource and
+	// not a tracked ARM resource.
+	proxyResource bool
+
+	// The Resource ID.
+	ID string `json:"id,omitempty"`
+
+	// The resource name.
+	Name string `json:"name,omitempty"`
+
+	// The resource type.
+	Type string `json:"type,omitempty" mutable:"case"`
+
+	// SystemData metadata relating to this resource.
+	SystemData *SystemData `json:"systemData,omitempty"`
+
+	// The SyncIdentityProvider Properties
+	Properties SyncIdentityProviderProperties `json:"properties,omitempty"`
+}
+
+// SyncSetProperties represents the properties of a SyncSet
+type SyncIdentityProviderProperties struct {
+	Resources string `json:"resources,omitempty"`
+}
+
+// SecretList represents a list of Secrets
+type SecretList struct {
+	// The list of syncsets.
+	Secrets []*Secret `json:"value"`
+
+	// The link used to get the next page of operations.
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+// Secret represents a secret.
+type Secret struct {
+	// This is a flag used during the swagger generation typewalker to
+	// signal that it should be marked as a proxy resource and
+	// not a tracked ARM resource.
+	proxyResource bool
+
+	// The Resource ID.
+	ID string `json:"id,omitempty"`
+
+	// The resource name.
+	Name string `json:"name,omitempty"`
+
+	// The resource type.
+	Type string `json:"type,omitempty" mutable:"case"`
+
+	// SystemData metadata relating to this resource.
+	SystemData *SystemData `json:"systemData,omitempty"`
+
+	// The Secret Properties
+	Properties SecretProperties `json:"properties,omitempty"`
+}
+
+// SecretProperties represents the properties of a Secret
+type SecretProperties struct {
+	// The Secrets Resources.
+	SecretResources string `json:"secretResources,omitempty"`
```

Comment:
nit: why is this one `NAMEResources` when others are just `Resources`?

Reply:
because of :drum: swagger generation. This resource is different than the others because it is a secretResource and needs to have a proper x-ms-secret header attached to it. This is a secret value which needs to be passed into the typewalker via the generator here https://github.com/Azure/ARO-RP/blob/42e7ef355f90d6c38cfe738d6fd9c0c8ff2826f8/pkg/swagger/generator.go#L127c

Reply:
therefore, since resources is overloaded I could not reuse that

Reply:
that is unfortunate :(